File: app/src/main/java/com/bccommittee/util/CurrencyUtil.java
package com.bccommittee.util;

import java.text.NumberFormat;
import java.util.Locale;

public class CurrencyUtil {
    private static final NumberFormat INR = NumberFormat.getCurrencyInstance(new Locale("en", "IN"));

    public static String format(long amount) {
        // e.g. ₹10,000
        return INR.format(amount).replace(".00", "");
    }

    public static String formatShort(long amount) {
        if (amount >= 100000) return "₹" + (amount / 100000) + "L";
        if (amount >= 1000)   return "₹" + (amount / 1000) + "K";
        return "₹" + amount;
    }
}

File: app/src/main/kotlin/com/khaga/ksociety/util/CurrencyUtil.kt
package com.khaga.ksociety.util

import java.text.NumberFormat
import java.util.Locale

object CurrencyUtil {

    private val inrFormat = NumberFormat.getCurrencyInstance(Locale("en", "IN"))

    /** Format as ₹10,000 */
    fun format(amount: Long): String =
        inrFormat.format(amount).replace(".00", "")

    /** Short format: ₹10K, ₹1L */
    fun formatShort(amount: Long): String = when {
        amount >= 100_000L -> "₹${amount / 100_000}L"
        amount >= 1_000L   -> "₹${amount / 1_000}K"
        else               -> "₹$amount"
    }
}
